web/src/hash_util.ts

```diff
@@ -325,3 +325,13 @@ export function decode_stream_topic_from_url(
         return null;
     }
 }
+
+export function is_search_view(): boolean {
+    const path = window.location.hash.split("/");
+    // The following check ensures the user has actually searched for something,
+    // and doesn't confuse it with the case in which the channel or topic name is "search".
+    if (path.length >= 2 && path.at(-2) === "search" && path.at(-1) !== "") {
+        return true;
+    }
+    return false;
+}
```

Comment:
No, you cannot parse the URL again.

Think a bit more. There's already logic, probably in `message_list_view.ts`, that determines whether to show individual cards for each message. How does that determine whether it's a search view? Your new logic should use the same method.

Reply:
ok